python_modules/dagster/dagster_tests/definitions_tests/test_asset_check_selection.py

```diff
@@ -0,0 +1,160 @@
+    asset_checks = [asset1_check1, asset1_check2, asset2_check1]
+    defs = Definitions(assets=assets, jobs=[asset_job], asset_checks=asset_checks)
+    job_def = defs.get_job_def(asset_job.name)
+    return job_def.execute_in_process()
+
+
+def test_job_with_all_checks_no_materializations():
+    job_def = define_asset_job("job1", selection=AssetSelection.all_asset_checks())
+    result = execute_asset_job_in_process(job_def)
+    assert result.success
+
+    assert len(result.get_asset_materialization_events()) == 0
+    check_evals = result.get_asset_check_evaluations()
+    assert {check_eval.asset_check_handle for check_eval in check_evals} == {
+        AssetCheckHandle(asset1.key, "asset1_check1"),
+        AssetCheckHandle(asset1.key, "asset1_check2"),
+        AssetCheckHandle(asset2.key, "asset2_check1"),
+    }
+
+
+def test_job_with_all_checks_for_asset():
+    job_def = define_asset_job("job1", selection=AssetSelection.asset_checks_for_assets(asset1))
+    result = execute_asset_job_in_process(job_def)
+    assert result.success
+
+    assert len(result.get_asset_materialization_events()) == 0
+    check_evals = result.get_asset_check_evaluations()
+    assert {check_eval.asset_check_handle for check_eval in check_evals} == {
+        AssetCheckHandle(asset1.key, "asset1_check1"),
+        AssetCheckHandle(asset1.key, "asset1_check2"),
+    }
+
+
+def test_job_with_asset_and_all_its_checks():
+    job_def = define_asset_job("job1", selection=AssetSelection.assets(asset1))
+    result = execute_asset_job_in_process(job_def)
+    assert result.success
+
+    assert len(result.get_asset_materialization_events()) == 1
+    check_evals = result.get_asset_check_evaluations()
+    assert {check_eval.asset_check_handle for check_eval in check_evals} == {
+        AssetCheckHandle(asset1.key, "asset1_check1"),
+        AssetCheckHandle(asset1.key, "asset1_check2"),
+    }
+
+
+def test_job_with_single_check():
+    job_def = define_asset_job("job1", selection=AssetSelection.asset_checks(asset1_check1))
+    result = execute_asset_job_in_process(job_def)
+    assert result.success
+
+    assert len(result.get_asset_materialization_events()) == 0
+    check_evals = result.get_asset_check_evaluations()
+    assert {check_eval.asset_check_handle for check_eval in check_evals} == {
+        AssetCheckHandle(asset1.key, "asset1_check1"),
+    }
+
+
+def test_job_with_all_assets_but_no_checks():
+    job_def = define_asset_job(
+        "job1", selection=AssetSelection.all() - AssetSelection.all_asset_checks()
+    )
+    result = execute_asset_job_in_process(job_def)
+    assert result.success
+
+    assert len(result.get_asset_materialization_events()) == 2
+    check_evals = result.get_asset_check_evaluations()
+    assert len(check_evals) == 0
+
+
+def test_job_with_asset_without_its_checks():
+    job_def = define_asset_job(
+        "job1", selection=AssetSelection.assets(asset1) - AssetSelection.all_asset_checks()
+    )
```

Comment:
Bit concerned about the discoverability of this one. I'm not sure how we are going to expect users to discover this. We may have to add additional helper methods for common use cases like this.